src/modm/ui/color/brightness.hpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2021, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_COLOR_BRIGHTNESS_HPP
+#define MODM_COLOR_BRIGHTNESS_HPP
+
+#include <stdint.h>
+
+#include <modm/io/iostream.hpp>
+#include <concepts>
+
+#include "hsv.hpp"
+#include "rgb.hpp"
+#include "rgb565.hpp"
+
+namespace modm
+{
+namespace color
+{
+
+// forward declarations for convertion constructors
+template<std::unsigned_integral T>
+class RgbT;
+
+template<std::unsigned_integral T>
+class HsvT;
+
+class Rgb565;
+
+/**
+ * Brightness, used with grayscale and monochrome displays, dimmed leds, brightness sensors
+ *
+ * \author		Thomas Sommer
+ * 
+ * \tparam	T	Unsigned Int for storing the brightness
+ * \ingroup		modm_ui_color
+ */
+
+template<std::unsigned_integral T = uint8_t>
+class BrightnessT
+{
+public:
+	T value{0};
+
+	constexpr BrightnessT() = default;
+
+	constexpr BrightnessT(T value) : value(value)
+	{}
+
+	/**
+	 * Convertion Constructor for RGB Color
+	 *
+	 * \param rgb	RGB Color
+	 */
+	template<std::unsigned_integral _T>
+	constexpr BrightnessT(RgbT<_T> rgb)
+		: value((0.2125 * float(rgb.red)) + (0.7154 * float(rgb.green)) + (0.0721 * float(rgb.blue)))
+	{}
+
+	/**
+	 * Convertion Constructor for HSV Color
+	 *
+	 * \param hsv	HSV Color
+	 */
+	template<std::unsigned_integral _T>
```

Comment:
```suggestion
	template<std::unsigned_integral U>
```

We usually use U, because it comes after T. `_` is reserved for system libs.